feat(test vectors): add basic and self-addressing

chunningham opened the pull request:
Adds some test vectors for inception events:
generated by keripy:
- self_addressing_1

generated by keriox:
- basic_1
- self_addressing_2

NOTE: currently the way signatures are created and verified for these implementations seems to be different. keriox uses the serialized extracted data form while keripy uses the raw serialized event message.

SmithSamuelM commented:
@chunningham 

Looks like there is a misunderstanding we need to resolve. I plan on addressing this in the meeting tomorrow.  

What I think the misunderstanding is that signatures and digests are on the full serialized events. The only place where the extracted data is used is to generate the derived self-addressing self-certifying prefix.     The prefix is then included in the event itself and the signature then wraps the event with the included identifier prefix. This is the most secure way.  We can talk about why generating signatures and digests from extracted data is less secure.  The design aesthetic for KERI is security first and implement with minimally sufficient means. This applies to flexibility or extensibility as well.  Don't make is as flexible as possible but only as flexible as necessary. And when there is conflict err on the side of security.  Don't vary from these principles just because you can but only because you must.

chunningham commented:
I see now where my misunderstanding lies :sweat_smile: I had interpreted the commitment to serialization to mean the inclusion of the vs string (and hence the type and size of the serialized event) in the SED, not to mean the use of the serialized event for signature gen/ver. I think at some point I made an assumption about the use of the SED and it stuck, but I'm satisfied by your explanation of the attack vector that introduces, thanks for it, I'll update keriox accordingly :+1:

chunningham commented:
updated the vectors to have correct signatures, and fixed an issue with the self-addressing identifier generation as well (length in the "vs" field was 0 instead of it's correct value, resulting in incorrect self-addressing identifiers)
